File: apps/main/src/types/conversation.ts
export interface SourceChunk {
  id: number;
  text: string;
  pageNumber: number;
  chapter?: string;
}

export interface Message {
  id: string;
  conversationId: string;
  role: 'user' | 'assistant';
  content: string;
  sourceChunks: SourceChunk[] | null;
  createdAt: number;
}

export interface Conversation {
  id: string;
  bookId: string;
  title: string;
  createdAt: number;
  updatedAt: number;
  messages: Message[];
}

File: apps/apple/Packages/RishiCore/Sources/RishiCore/Models/Conversation.swift
import Foundation

public struct Conversation: Codable, Sendable, Hashable, Identifiable {
    public let id: ConversationID
    public var userId: UserID
    public var bookId: BookID?
    public var title: String
    public var createdAt: Date
    public var updatedAt: Date

    public init(
        id: ConversationID = UUID(),
        userId: UserID,
        bookId: BookID? = nil,
        title: String,
        createdAt: Date = Date(),
        updatedAt: Date = Date()
    ) {
        self.id = id
        self.userId = userId
        self.bookId = bookId
        self.title = title
        self.createdAt = createdAt
        self.updatedAt = updatedAt
    }
}
